src/Compose/DeveloperBuilder.php

```diff
@@ -92,45 +84,22 @@ public function build(Config $config): Manager
             ->build($config);
 
         $syncEngine = $config->getSyncEngine();
-        $syncConfig = [];
-
-        if ($syncEngine === self::SYNC_ENGINE_DOCKER_SYNC) {
-            $syncConfig = ['external' => true];
-        } elseif ($syncEngine === self::SYNC_ENGINE_MUTAGEN) {
-            $syncConfig = [];
-        } elseif ($syncEngine === self::SYNC_ENGINE_NATIVE) {
-            $syncConfig = [
-                'driver_opts' => [
-                    'type' => 'none',
-                    'device' => $this->resolver->getRootPath(),
-                    'o' => 'bind'
-                ]
-            ];
-        }
-
         $volumesList = [
-            $volumePrefix . self::VOLUME_MAGENTO_SYNC => $syncConfig,
             $volumePrefix . self::VOLUME_MAGENTO_DB => []
         ];
 
+        if (in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)) {
+            $volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncEngine === self::SYNC_ENGINE_DOCKER_SYNC
+                ? ['external' => true]
+                : [];
+        }
```

Comment:
```suggestion
        $volumes = [self::VOLUME_MAGENTO . ':' . self::DIR_MAGENTO . ':delegated'];

        if (in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)) {
            $volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncEngine === self::SYNC_ENGINE_DOCKER_SYNC
                ? ['external' => true]
                : [];
            $volumes = [$volumePrefix . self::VOLUME_MAGENTO_SYNC . ':' . self::DIR_MAGENTO . ':nocopy'];
        }
```

And remove the method `getMagentoVolumes()`